xarray/core/dataset.py

```diff
@@ -915,7 +915,9 @@ def copy(self: T, deep: bool = False, data: Mapping = None) -> T:
             variables = OrderedDict((k, v.copy(deep=deep, data=data.get(k)))
                                     for k, v in self._variables.items())
 
-        return self._replace(variables)
+        attrs = copy.deepcopy(self._attrs) if deep else copy.copy(self._attrs)
```

Comment:
I made one small fix, renaming `_attrs` to `attrs`.

As a general rule, we use the underscore prefix in Python as a convention for "private" but technically public variables. This isn't necessary for local variables defined within the scope of a function, since these are already private.

Reply:
Right. Actually I first called it "new_attrs" but it was 3 characters above PEP8 line-length limit, so I removed the first 3... :)
Thanks for the fix.